spec/datadog/core/environment/execution_spec.rb

```diff
@@ -114,5 +118,94 @@
         end
       end
     end
+
+    context 'when webmock has enabled net-http adapter' do
+      before do
+        allow(described_class).to receive(:repl?).and_return(false)
+        allow(described_class).to receive(:test?).and_return(false)
+        allow(described_class).to receive(:rails_development?).and_return(false)
+
+        WebMock.enable!
+      end
+
+      it { is_expected.to eq(true) }
+    end
+  end
+
+  describe '.webmock_enabled?' do
+    context 'when missing constant `WebMock`' do
+      it do
+        hide_const('::WebMock')
+        expect(described_class).not_to be_webmock_enabled
+      end
+    end
+
+    context 'when missing constant `Net::HTTP`' do
+      it do
+        hide_const('::Net::HTTP')
+        expect(described_class).not_to be_webmock_enabled
+      end
+    end
+
+    context 'when `WebMock` and `Net::HTTP` constants both exist' do
+      before do
+        WebMock.enable!
+      end
+
+      it do
+        expect(described_class).to be_webmock_enabled
+      end
+
+      it do
+        WebMock.enable!(except: [:net_http])
+
+        expect(described_class).not_to be_webmock_enabled
+      end
+
+      it do
+        WebMock.disable!
+
+        expect(described_class).not_to be_webmock_enabled
+      end
+
+      it do
+        WebMock.disable!(except: [:net_http])
+
+        expect(described_class).to be_webmock_enabled
+      end
+    end
+
+    [1, 2, 3].each do |version|
+      context "when given WebMock version #{version}.x", skip: Gem::Version.new(Bundler::VERSION) < Gem::Version.new('2') do
+        it do
+          out, err = Bundler.with_clean_env do
+            Open3.capture3('ruby', stdin_data: <<-RUBY
+              require 'bundler/inline'
+
+              gemfile(true, quiet: true) do
+                source 'https://rubygems.org'
+                gem 'webmock', "~> #{version}"
+              end
+
+              require 'webmock'
+              WebMock.enable!
+
+              lib = File.expand_path('lib', __dir__)
+              $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+              require 'datadog/core/environment/execution'
+
+              STDOUT.print "WebMock::VERSION:"
+              STDOUT.print WebMock::VERSION
+              STDOUT.print "=>"
```

Comment:
Less code to maintain 🤷 
```suggestion
              STDOUT.print "WebMock::VERSION:#{WebMock::VERSION}"
```